alerter.go

```diff
@@ -111,36 +111,38 @@ func ReportErrors(config Config, errors []string) []string {
 	return report
 }
 
-func included(logMessage string, includes []string) bool {
-	if len(includes) == 0 {
-		return true
-	}
-
-	for _, include := range includes {
-		matched, err := regexp.MatchString(include, logMessage)
-		if err != nil {
-			panic(err)
-		}
+type PatternMatcher struct {
+	includes []*regexp.Regexp
+	excludes []*regexp.Regexp
+}
 
-		if matched {
-			return true
+func NewPatternMatcher(includes []string, excludes []string) *PatternMatcher {
+	toRegexp := func(patterns []string) []*regexp.Regexp {
+		allRegexp := make([]*regexp.Regexp, 0, len(patterns))
+		for _, pattern := range patterns {
+			allRegexp = append(allRegexp, regexp.MustCompile(pattern))
 		}
+		return allRegexp
 	}
-	return false
+	return &PatternMatcher{includes: toRegexp(includes), excludes: toRegexp(excludes)}
 }
 
-func excluded(logMessage string, excludes []string) bool {
-	for _, exclude := range excludes {
-		matched, err := regexp.MatchString(exclude, logMessage)
-		if err != nil {
-			panic(err)
+func (matcher *PatternMatcher) FindAllMatch(s string) []string {
+	matches := make([]string, 0, 0)
+	for _, include := range matcher.includes {
+		for _, match := range include.FindAllString(s, -1) {
+			matches = append(matches, match)
 		}
+	}
 
-		if matched {
-			return true
+	for _, exclude := range matcher.excludes {
+		for i, match := range matches {
+			if len(exclude.FindAllString(match, -1)) != 0 {
+				matches = append(matches[:i], matches[i+1:]...)
```

Comment:
@totahuanocotl Good point. Probably need to do somehting like that http://stackoverflow.com/questions/20545743/delete-entries-from-a-slice-while-iterating-over-it-in-go
Will pick it up when im back, unless u want to pick it up :-D